pmlc/rt/jit_executable.cc

```diff
@@ -73,89 +73,41 @@ static void setupTargetTriple(llvm::Module *llvmModule) {
   llvmModule->setTargetTriple(targetTriple);
 }
 
-static std::string makePackedFunctionName(StringRef name) {
-  return "_mlir_" + name.str();
-}
-
-static std::string makeCWrapperFunctionName(StringRef name) {
-  return "_mlir_ciface_" + name.str();
-}
-
-// Define an interface function that wraps all the arguments of the original
-// function and all its results into an i8** pointer to provide a unified
-// invocation interface.
-static void packFunctionArguments(llvm::Module *module, StringRef entry) {
-  auto &ctx = module->getContext();
-  llvm::IRBuilder<> builder(ctx);
-  auto funcName = makeCWrapperFunctionName(entry);
-  auto *func = module->getFunction(funcName);
-  if (!func) {
-    throw std::runtime_error("Could not find function: " + funcName);
-  }
-
-  // Given a function `foo(<...>)`, define the interface function
-  // `mlir_foo(i8**)`.
-  auto newType = llvm::FunctionType::get(builder.getVoidTy(),
-                                         builder.getInt8PtrTy()->getPointerTo(),
-                                         /*isVarArg=*/false);
-  auto newName = makePackedFunctionName(entry);
-  auto funcCst = module->getOrInsertFunction(newName, newType);
-  llvm::Function *interfaceFunc = cast<llvm::Function>(funcCst.getCallee());
-
-  // Extract the arguments from the type-erased argument list and cast them to
-  // the proper types.
-  auto bb = llvm::BasicBlock::Create(ctx);
-  bb->insertInto(interfaceFunc);
-  builder.SetInsertPoint(bb);
-  llvm::Value *argList = interfaceFunc->arg_begin();
-  SmallVector<llvm::Value *, 8> args;
-  args.reserve(llvm::size(func->args()));
-  for (auto &indexedArg : llvm::enumerate(func->args())) {
-    llvm::Value *argIndex = llvm::Constant::getIntegerValue(
-        builder.getInt64Ty(), APInt(64, indexedArg.index()));
-    llvm::Value *argPtrPtr = builder.CreateGEP(argList, argIndex);
-    llvm::Value *argPtr = builder.CreateLoad(argPtrPtr);
-    llvm::Value *arg =
-        builder.CreateBitCast(argPtr, indexedArg.value().getType());
-    args.push_back(arg);
-  }
-
-  // Call the implementation function with the extracted arguments.
-  builder.CreateCall(func, args);
-
-  // The interface function returns void.
-  builder.CreateRetVoid();
-}
-
 namespace {
 
-class MemRefDescriptor {
-private:
-  struct Base {
+class MemRef {
+public:
+  struct Descriptor {
```

Comment:
What do we gain by refactoring as such? This seems confusing to me, we want to manipulate a `MemRefDescriptor` not a `MemRef` itself (a memref is just a logical concept).